core/src/main/scala/org/apache/spark/sql/execution/columnar/ColumnTableScan.scala

```diff
@@ -801,10 +802,10 @@ private[sql] final case class ColumnTableScan(
         val nullCheckAddon =
           if (notNullVar != null) s"if ($notNullVar < 0) $nullVar = $col == null;\n"
           else ""
-        assignCode =
+        stringAssignCode =
             s"($dictionary != null ? $dictionary[$dictIndex] " +
-                s": $decoder.readUTF8String($buffer, $cursorVar));\n" +
-                s"$nullCheckAddon"
+                s": $decoder.readUTF8String($buffer, $cursorVar));\n"
+        assignCode = stringAssignCode + s"$nullCheckAddon"
```

Comment:
Don't need s"" if nullCheckAddon is only value in string and just a '+' of two will do